src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/BeginUpdateInstance.sql

```diff
@@ -0,0 +1,49 @@
+﻿/*************************************************************
+    Stored procedures for updating an instance status.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     BeginUpdateInstance
+--
+-- DESCRIPTION
+--     Updates a DICOM instance NewWatermark
+--
+-- PARAMETERS
+--     @partitionKey
+--         * The system identified of the data partition.
+--     @watermarkTableType
+--         * The SOP instance watermark.
+CREATE OR ALTER PROCEDURE dbo.BeginUpdateInstance
+	@partitionKey       INT,
+	@watermarkTableType dbo.WatermarkTableType READONLY
+AS
+BEGIN
+    SET NOCOUNT ON
+    SET XACT_ABORT ON
+
+    BEGIN TRANSACTION
+        
+        UPDATE i
+        SET NewWatermark = NEXT VALUE FOR dbo.WatermarkSequence
+        FROM dbo.Instance i
+        JOIN @watermarkTableType input ON  i.Watermark = input.Watermark AND i.PartitionKey = @partitionKey
+        WHERE Status = 1
+
+        -- The instance does not exist.
+        IF @@ROWCOUNT = 0
```

Comment:
what if you had 10 rows in watemarkTable and you just update say 5 rows? See if it makes sense to keep these checks?

Reply:
I kept these checks thinking the instance could be deleted while in update operation.

Reply:
Not getting your point on just updating 5 rows. What is the scenario that could happen?

Reply:
This is removed